In [ ]:
import numpy as np


X = np.array([0.5, 0.75, 1, 1.25, 1.5, 1.75, 1.75, 2, 2.25, 2.5, 2.75, 3, 3.25, 3.5, 4, 4.25, 4.5, 4.75, 5, 5.5]).reshape(-1,1)
y = np.array([0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 1, 1, 1, 1])

In [ ]:
from sklearn.linear_model import LogisticRegression

regresion_logistica = LogisticRegression()


regresion_logistica.fit(X,y)

LogisticRegression()

In [ ]:
X_nuevo = np.array([1, 2, 3, 4, 5, 6]).reshape(-1,1)

# Paso 6: Usa el modelo entrenado para obtener las predicciones con datos nuevos
prediccion = regresion_logistica.predict(X_nuevo)
print(prediccion)
# produce el resultado: [0 0 1 1 1 1]

[0 0 1 1 1 1]


Estudiar 1 o 2 horas implica suspender, estudiar 3 o más aprobar

In [ ]:
probabilidades_prediccion = regresion_logistica.predict_proba(X_nuevo)
print(probabilidades_prediccion)

[[0.87984046 0.12015954]
 [0.69895285 0.30104715]
 [0.42402118 0.57597882]
 [0.18924992 0.81075008]
 [0.06891381 0.93108619]
 [0.02293029 0.97706971]]


In [ ]:
print(probabilidades_prediccion[:,1])

[0.12015954 0.30104715 0.57597882 0.81075008 0.93108619 0.97706971]


Estudiando 1 hora la probable de aprobar es de 12%, estudiando 6 horas la probabilidad de aprobar es de 97%